examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairingCommand.kt

```diff
@@ -0,0 +1,253 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package com.matter.controller.commands.pairing
+
+import chip.devicecontroller.ChipDeviceController
+import com.matter.controller.commands.common.CredentialsIssuer
+import com.matter.controller.commands.common.IPAddress
+import com.matter.controller.commands.common.MatterCommand
+import java.net.InetAddress
+import java.net.UnknownHostException
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicLong
+import java.util.logging.Level
+import java.util.logging.Logger
+
+abstract class PairingCommand(
+  controller: ChipDeviceController,
+  commandName: String?,
+  mode: PairingModeType,
+  network: PairingNetworkType,
+  credsIssuer: CredentialsIssuer?,
+  filter: DiscoveryFilterType
+) : MatterCommand(controller, commandName, credsIssuer), ChipDeviceController.CompletionListener {
+  private val logger = Logger.getLogger(PairingCommand::class.java.name)
+  private var pairingMode: PairingModeType = PairingModeType.NONE
+  private var networkType: PairingNetworkType = PairingNetworkType.NONE
+  private var filterType = DiscoveryFilterType.NONE
+  private var remoteAddr: IPAddress? = null
+  private val nodeId: AtomicLong = AtomicLong()
+  private val discoveryFilterCode: AtomicLong = AtomicLong()
+  private val timeoutMillis: AtomicLong = AtomicLong()
+  private val discoverOnce: AtomicBoolean = AtomicBoolean()
+  private val useOnlyOnNetworkDiscovery: AtomicBoolean = AtomicBoolean()
+  private val remotePort: AtomicInteger = AtomicInteger()
+  private val discriminator: AtomicInteger = AtomicInteger()
+  private val setupPINCode: AtomicLong = AtomicLong()
+  private val operationalDataset = StringBuffer()
+  private val SSID = StringBuffer()
+  private val password = StringBuffer()
+  private val onboardingPayload = StringBuffer()
+  private val discoveryFilterInstanceName = StringBuffer()
+
+  constructor(
+    controller: ChipDeviceController,
+    commandName: String?,
+    mode: PairingModeType,
+    network: PairingNetworkType,
+    credsIssuer: CredentialsIssuer?
+  ) : this(controller, commandName, mode, network, credsIssuer, DiscoveryFilterType.NONE)
+
+  init {
+    pairingMode = mode
+    networkType = network
+    filterType = filter
+    try {
```

Comment:
If you are rethrowing anyway, you can remove the try/catch since Kotlin does not enforce exception checking:

```
private val remoteAddr = IPAddress(InetAddress.getByName("::1"))
```